### DISCUSSION TITLE
log ebus messages to a log file using An ebusd add-on

### DISCUSSION THREAD

**Discussion by FrigerioSergio** [john30/ebusd#595] [Q&A]
Home Assistant integration [An ebusd add-on] (https://community.home-assistant.io/t/an-ebusd-add-on/344852)

I do not understand how to save the full ebus message log to a a file.......

Any suggestion?

**Comment by tim-devel** [comment]
Try adding this to your config:

--logfile=/config/ebusd/ebusd.log

**Reply by FrigerioSergio** [reply]
Hello,
--logfile=/config/ebusd/ebusd.log is not recognized inside ebusd configuration table.

I have tried with this:
![image](https://user-images.githubusercontent.com/22483463/166154735-b6723c93-137f-41a1-8ec8-22e4584eb3e3.png)

but I got this error:
22-05-01 18:12:20 WARNING (MainThread) [supervisor.addons.options] Option 'logfile' does not exist in the schema for eBUSd (2ad9b828_ebusd)

Any Idea?

Thanks and best regards,

Sergio

**Reply by tim-devel** [reply]
Try:

commandline_options: --logfile=/config/ebusd/ebusd.log

**Reply by FrigerioSergio** [reply]
Hello, i did not get any file....

This is what I have done:

![image](https://user-images.githubusercontent.com/22483463/167144676-0d184c0c-645c-4389-a186-43fa69d05ef9.png)

This is what I got in the configuration YAML:

![image](https://user-images.githubusercontent.com/22483463/167144947-457a44dc-1fb8-4631-a5f9-a8039b8a0e5a.png)

This is the EBUSD register:

![image](https://user-images.githubusercontent.com/22483463/167145239-e27a82c3-3002-4715-b648-fd27e6539619.png)
 
Any Idea? Thanks!

**Reply by tim-devel** [reply]
Ah, it's because the add-on doesnt have write access to that folder.  I'll do a new release of the add-on with write access enabled

**Reply by tim-devel** [reply]
Just tested and logfile option only works in deamon mode.  Ebusd runs in foreground in the addon

**Reply by FrigerioSergio** [reply]
Ok thanks!

**Reply by john30** [reply]
if started in foreground, there is indeed no log file as everathing is written to stdout/stderr. using `docker logs` would allow saving the output to a file. maybe HA already allows downloading the output from UI or if not such a feature might be helpful for other supervised containers as well and someone might want to add a feature request to HA?

**Comment by ccle610** [comment]
Ah superdooper !
Since today, there is an update, writing S6 Logs into /config (can be configured directly via AddOn)

Thank You so much!
